### PR TITLE
fix(curriculum): improve test robustness

### LINKED ISSUE #38906: Make A Person Test Suite Fragile
**Describe your problem and how to reproduce it:**
The test suite for 'Make A Person' is a little fragile and needs some `// Only change...` comments.

The below code will fail the test suite even though the object definition is correct because a user changed the code outside of the object definition:
```
var Person = function(firstAndLast) {
  const names = firstAndLast.split(' ');
  let firstName = names[0];
  let lastName = names[1];
  // Complete the method below and implement the others similarly
  // Getters
  this.getFirstName = function() {
    return firstName;
  };
  this.getLastName = function() {
    return lastName;
  };
  this.getFullName = function() {
    return firstName + " " + lastName;
  };
  // Setters
  this.setFirstName = function(name) {
    firstName = name;
  };
  this.setLastName = function(name) {
    lastName = name;
  }
  this.setFullName = function(name) {
    const names = name.split(' ');
    firstName = names[0];
    lastName = names[1];
  }
};

var bob = new Person('Bob Ross');
bob.getFullName();
bob.setFirstName("Haskell") // This code passes if this line is removed
```

A fix is to add the `// Only change...` comments

New starter code:
```
var Person = function(firstAndLast) {
  // Only change code below this line
  // Complete the method below and implement the others similarly
  this.getFullName = function() {
    return "";
  };
  return firstAndLast;
  // Only change code above this line
};

var bob = new Person('Bob Ross');
bob.getFullName();
```

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person

**Tell us about your browser and operating system:**
* Browser Name: Firefox
* Browser Version: 76.0.1 (64-bit)
* Operating System: Linux

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
image

### PR DESCRIPTION
**Pull request by kushagra010**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38906

### CODE REVIEW

**Comment by kushagra010** [comment]
Hi,
I am a beginner and wanted to contribute to this issue, I made the required changes. Please review it @ojeytonwilliams.
Thankyou.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
HI @kushagra010 thanks for the PR.

While we did consider adding comments to the code, we decided on a different solution. We can allow the user to modify `bob` if we add to the `<section id='challengeSeed'>` by including 

````md
### After Test
<div id='js-teardown'>

```js
if (bob) {
  bob = new Person("Bob Ross");
}
```
</div>
````

could you test that out and, if it works as desired, update the PR?

**Comment by kushagra010** [comment]
After the rquired changes , the test was not successful, can you help me out @ojeytonwilliams.

**Comment by ojeytonwilliams** [comment]
I'd take a look at the contributing guidelines - there's a [section](https://contribute.freecodecamp.org/#/how-to-work-on-coding-challenges?id=challenge-template) on the structure of the challenges that should help you out.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md [hunk 1]
```diff
@@ -83,6 +85,16 @@ bob.getFullName();
...
+### Before Test
+
+<div id='js-setup'>
+
+```js
+if(bob){
+  bob = new Person("Bob Ross");
+}
+```
+</div>
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md:97]:
```suggestion

</div>
```
To fix the lint error.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md [hunk 2]
```diff
@@ -70,11 +70,13 @@ tests:
 
 ```js
 var Person = function(firstAndLast) {
+  // Only change code below this line
   // Complete the method below and implement the others similarly
   this.getFullName = function() {
     return "";
-  };
+  };
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md:77]:
```suggestion
  };
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md [hunk 3]
```diff
@@ -70,11 +70,13 @@ tests:
...
 ```js
 var Person = function(firstAndLast) {
+  // Only change code below this line
   // Complete the method below and implement the others similarly
   this.getFullName = function() {
     return "";
-  };
+  }; 
   return firstAndLast;
+  // Only change code above this line
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md:79]:
```suggestion
```
As discussed in the issue, there's no need to heavily restrict what the user changes.  As long as they keep

```js
var Person = function(firstAndLast) {
```

unchanged they can alter whatever they want.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Comment by kushagra010** [comment]
Thankyou @ojeytonwilliams , for your guidance.
I have made the required changes, please review it.

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1: 

Nice work @kushagra010!

**Comment by kushagra010** [comment]
Thankyou,
So what about merging now. Since this would be my first merging, so I am bit excited about it.

**Comment by ojeytonwilliams** [comment]
Typically we merge after two reviews, in case the first reviewer misses something.  I've let the moderators know that this is ready for a final review. They are volunteers with other demands on their time, though, so I can't tell you exactly when that will happen.

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
@kushagra010  The check for the `bob` object should be placed in the "After Test" section, so that it can be run once the user's code has been submitted -
```
### After Test

<div id='js-teardown'>

```js
if(bob){
  bob = new Person("Bob Ross");
}

</div>
```

Note - Do add the closing set of backticks after the `if` block, I couldn't put it here because it wouldn't format the code section correctly otherwise.

**Comment by kushagra010** [comment]
I have done the suggested changes. Please review it @Manish-Giri

**Review by Manish-Giri — APPROVED** [review]
LGTM 💯 

@ojeytonwilliams If you could take a final look and merge if everything sounds good.

**Comment by ojeytonwilliams** [comment]
@kushagra010  Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Comment by kushagra010** [comment]
Thank you @ojeytonwilliams for your guidance.

**Pull request merged**